### PR TITLE
Adding configmap-reloader sidecar to prometheus

### PR DESCRIPTION
**Pull request by jpweber**
## Description

This PR adds a sidecar to prometheus to allow reloading the config of prometheus when the configmap for the configuration or alert rules is modified. 

## 🎟 Issue(s)

Resolves https://github.com/astronomer/issues/issues/1832

## 🧪  Testing

There is risk of the configmap-reloader not triggering, or not communicating correctly with prometheus. A unit test has been added to test for the presence of the sidecar. A functional test has been added that modifies the configmap and looks for those changes to be reflected in the container. 

- [x] The PR title is informative to the user experience

### CODE REVIEW

File: bin/test-ap [hunk 1]
```diff
@@ -16,9 +16,5 @@ pip install -r $REPO_DIR/bin/functional-tests/requirements.txt
 export NAMESPACE=astronomer
 export RELEASE_NAME=astronomer
 
-# Sleep twice the prometheus scrape period
-# plus five seconds to allow for metrics to be
-# collected, which the functional tests need.
-sleep 65
```
Comment (sjmiller609) [inline bin/test-ap:22]: I moved your test to the top and deleted this sleep

**Comment by jpweber** [comment]
Making some changes to improve the functional tests. Moving to draft until that is complete.

File: bin/functional-tests/test_config.py [hunk 2]
```diff
@@ -11,6 +11,68 @@
...
 import os
 import docker
+import time
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+
+# This test should remain first because this sleep is required to allow for other tests
+# to have enough time for metrics to be collected.
+def test_prometheus_config_reloader_works(prometheus, kube_client):
+    """ Ensure that Prometheus reloads it's config when the cofigMap is updated
```
Comment (andriisoldatenko) [inline bin/functional-tests/test_config.py:21]:
```suggestion
    """ 
    Ensure that Prometheus reloads it's config when the cofigMap is updated
    ...
    """
```

Comment (danielhoherd) [reply bin/functional-tests/test_config.py:21]: This is why we need `black`

**Comment by jpweber** [comment]
Ok, test has been modified to work better and more accurately. I also took andrii's suggestion and committed.

File: bin/functional-tests/test_config.py [hunk 3]
```diff
@@ -129,3 +71,75 @@ def test_houston_metrics_are_collected(prometheus):
     parsed = json.loads(data)
     assert len(parsed['data']['result']) > 0, \
         f"Expected to find a metric houston_up, but we got this response:\n\n{parsed}"
+
+
+def test_prometheus_config_reloader_works(prometheus, kube_client):
+    """ 
+    Ensure that Prometheus reloads it's config when the cofigMap is updated
```
Comment (danielhoherd) [inline bin/functional-tests/test_config.py:78]:
```suggestion
    Ensure that Prometheus reloads its config when the cofigMap is updated
```

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**